### ISSUE TITLE
Improve Loading / Error Substates page

### ISSUE THREAD

**Issue by Dremora** [emberjs/guides#328]
[This page](http://guides.emberjs.com/v1.11.0/routing/loading-and-error-substates/) talks about `loading` event and has a call to `displayLoadingSpinner()` included in two of its snippets. This is exactly what I'm trying to do — however, there's no mention on how this function could be implemented (not to mention is should probably be a method). I can't figure out how to access controller from within `loading` event handler.

The documentation should be improved to provide relevant information. If it's impossible to change controller's state from the event handler, the mention of `displayLoadingSpinner()` should be removed.

**Comment by michaelrkn** [comment]
Agreed. @Dremora did you figure out how to do this? If so, it would be awesome if you could open a PR to help explain it.

**Comment by locks** [comment]
Getting a hold of the controller should be as simple as `this.controller`. The `displayLoadingSpinner` function is left as an exercise to the reader because it could be anything. It being a "global" function is an artefact from the pre-global days, I'd take suggestions as how to improve it.

Would something like:

```
export default Ember.Route.extend({
  displayLoadingSpinner() {
    this.controller.set('isLoading', true);
  },

  actions: {
    loading: function(transition, originRoute) {
      this.displayLoadingSpinner();

      // substate implementation when returning `true`
      return true;
    }
  }
});
```

solve your problem?

**Comment by JKGisMe** [comment]
It also is not complete. According to this issue: https://github.com/emberjs/ember.js/issues/11888 the examples in the guides would never trigger the loading substate
```javascript
Router.map(function() {
  this.route('foo', function() {
    this.route('bar', function() {
      this.route('baz');
    });
  });
});
```

It would have to be:

```javascript
Router.map(function() {
  this.route('foo', function() {
    this.route('bar', function() {
      this.route('baz', function(){});
    });
  });
});
```

And according to this issue: https://github.com/emberjs/ember.js/issues/10645 when using pods the loading/error templates have to actually be put in separate directories which is never mentioned in the guides.

```
baz/controller.js
baz/route.js
baz/template.js
baz-loading/template.hbs
baz-error/template.hbs
```

As more people start to use pods this can be really hard to figure out. Not that I spent an inordinate amount of time trying to figure it out or anything.

And for the comment before mine... `this.controller` is private.

**Comment by michaelrkn** [comment]
Fixed by #699.
